### DISCUSSION TITLE
[request] support to build gumbo with system library

### DISCUSSION THREAD

**Discussion by darix** [sparklemotion/nokogiri#2319] [RFCs]
This works for all other libraries. is the intree copy so heavily patched that it can not just use the system copy?

**Comment by darix** [comment]
so after some patching to try to build with the system copy ... it seems like the answer is "yes heavily patched"

which brings us to the question ... what blocks upstreaming those changes?

**Comment by stevecheckoway** [comment]
[Gumbo](https://github.com/google/gumbo-parser) is completely unmaintained. I think it was started as someone's 20% time project at Google and that person has left. It hasn't been updated in years. I'd be happy to upstream them (although at this point, a _lot_ has changed) if there were anyone at Google who wanted to accept the changes.

**Comment by darix** [comment]
there are other project that use the gumbo-parser like mupdf and claws-mail.

maybe time for a fork? lets see if we can poke some people at google.

**Comment by stevecheckoway** [comment]
It has some serious bugs in it. I wouldn't use the official version on any untrusted input.

**Comment by flavorjones** [comment]
I'm going to convert this to a "discussion" since there's nothing immediately actionable, but I want to encourage a conversation about upstreaming.

**Comment by darix** [comment]
JFYI https://twitter.com/arw/status/1626016312467271682

so the gumbo-parser project is officially declared dead.

**Comment by txgk** [comment]
Hello, I felt the whole spectrum of your feelings and decided to start maintaining gumbo-parser over Codeberg https://codeberg.org/grisha/gumbo-parser. I will be very happy to get any help and pull requests. Already sent messages to package maintainers about the creation of a new maintained project.

I will leave a copy of the message here:
```
Hello, dear gumbo-parser package maintainer.

I write to you because old gumbo-parser repository has been archived and marked
unmaintained by original author (actually it was unmaintained since 2016).

Fortunately I created fork which continues the ideas of the original project,
monitors bug reports and is being maintained. It's hosted on Codeberg (grisha/gumbo-parser).

In the new release of gumbo-parser 0.11.0 I added support for picture tag and
made one of the build script compatible with Python 3. Also all email addresses
for bug reporting were changed to my email address.

Main reasoning behind this move is my use of gumbo-parser in one of the software
projects, which strives to develop and unmaintained gumbo-parser is a block on
the way to this.

I humbly ask you to update gumbo-parser to 0.11.0 changing source repo to page on
Codeberg (grisha/gumbo-parser). Thank you very much :)

Best regards, Grigory Kirillov
```
